OpenRA.Platforms.Default/Sdl2GraphicsDevice.cs

```diff
@@ -98,6 +86,47 @@ public Sdl2GraphicsDevice(Size windowSize, WindowMode windowMode)
 
 			SDL.SDL_SetModState(SDL.SDL_Keymod.KMOD_NONE);
 			input = new Sdl2Input();
+
+			Clear();
+			Present();
+			SetWindowSize(WindowSize, windowMode);
+		}
+
+		public void SetWindowSize(Size windowSize, WindowMode windowMode)
+		{
+			VerifyThreadAffinity();
+
+			SDL.SDL_DisplayMode display;
+
+			// SDL2 doesn't support resolution switching whilst in full screen mode.
+			SDL.SDL_SetWindowFullscreen(window, 0);
+			SDL.SDL_GetCurrentDisplayMode(0, out display);
+
+			if (windowMode == WindowMode.NativeFullscreen)
+				windowSize = new Size(display.w, display.h);
+
+			SDL.SDL_SetWindowSize(window, windowSize.Width, windowSize.Height);
+			SDL.SDL_SetHint(SDL.SDL_HINT_VIDEO_MINIMIZE_ON_FOCUS_LOSS, windowMode == WindowMode.Fullscreen ? "1" : "0");
+
+			if (windowMode == WindowMode.Fullscreen)
+			{
+				display.w = windowSize.Width;
+				display.h = windowSize.Height;
+				SDL.SDL_SetWindowDisplayMode(window, ref display);
+				SDL.SDL_SetWindowFullscreen(window, (uint)SDL.SDL_WindowFlags.SDL_WINDOW_FULLSCREEN);
+			}
+
+			if (windowMode == WindowMode.NativeFullscreen)
+				SDL.SDL_SetWindowFullscreen(window, (uint)SDL.SDL_WindowFlags.SDL_WINDOW_FULLSCREEN_DESKTOP);
+
+			if (windowMode == WindowMode.Windowed)
+				SDL.SDL_SetWindowPosition(window, SDL.SDL_WINDOWPOS_CENTERED, SDL.SDL_WINDOWPOS_CENTERED);
+			else
+				SDL.SDL_SetWindowPosition(window, 0, 0);
```

Comment:
This is no longer a workaround but required by all systems. Should I still grab the comment?